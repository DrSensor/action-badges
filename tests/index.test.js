const request = require('supertest')
const nock = require('nock')
const createServer = require('../lib')

describe('action-badges', () => {
  let app

  beforeEach(() => {
    app = createServer()
    console.error = jest.fn()
  })

  it('has a test', () => {
    expect(1 + 1).toBe(2)
  })

  describe('GET /', () => {
    it('returns the homepage', async () => {
      const res = await request(app).get('/')
      expect(res.status).toBe(200)
      expect(res.text).toMatchSnapshot()
    })
  })

  describe('/:owner/:repo', () => {
    it('returns some SVG code', async () => {
      nock('https://api.github.com')
        .get(/\/repos\/JasonEtco\/example\/commits\/master\/check-suites/)
        .reply(200, { check_suites: [{ conclusion: 'success' }] })

      const res = await request(app).get('/JasonEtco/example')
      expect(res.status).toBe(200)
      expect(res.body.toString('utf8')).toMatchSnapshot()
    })

    it('returns an "error" badge on errors', async () => {
      nock('https://api.github.com')
        .get(/\/repos\/JasonEtco\/example\/commits\/master\/check-suites/)
        .reply(500)

      const res = await request(app).get('/JasonEtco/example')
      expect(res.status).toBe(200)
      expect(res.body.toString('utf8')).toMatchSnapshot()
    })
  })
})
