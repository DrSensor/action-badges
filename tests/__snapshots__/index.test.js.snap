// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action-badges /:owner/:repo returns an "error" badge on errors 1`] = `
"<svg width=\\"74.2\\" height=\\"20\\" viewBox=\\"0 0 742 200\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <linearGradient id=\\"a\\" x2=\\"0\\" y2=\\"100%\\">
    <stop offset=\\"0\\" stop-opacity=\\".1\\" stop-color=\\"#EEE\\"/>
    <stop offset=\\"1\\" stop-opacity=\\".1\\"/>
  </linearGradient>
  <mask id=\\"m\\"><rect width=\\"742\\" height=\\"200\\" rx=\\"30\\" fill=\\"#FFF\\"/></mask>
  <g mask=\\"url(#m)\\">
    <rect width=\\"368\\" height=\\"200\\" fill=\\"#555\\"/>
    <rect width=\\"374\\" height=\\"200\\" fill=\\"#E43\\" x=\\"368\\"/>
    <rect width=\\"742\\" height=\\"200\\" fill=\\"url(#a)\\"/>
  </g>
  <g fill=\\"#fff\\" text-anchor=\\"start\\" font-family=\\"Verdana,DejaVu Sans,sans-serif\\" font-size=\\"110\\">
    <text x=\\"60\\" y=\\"148\\" textLength=\\"268\\" fill=\\"#000\\" opacity=\\"0.25\\">build</text>
    <text x=\\"50\\" y=\\"138\\" textLength=\\"268\\">build</text>
    <text x=\\"423\\" y=\\"148\\" textLength=\\"274\\" fill=\\"#000\\" opacity=\\"0.25\\">error</text>
    <text x=\\"413\\" y=\\"138\\" textLength=\\"274\\">error</text>
  </g>
  
</svg>"
`;

exports[`action-badges /:owner/:repo returns some SVG code 1`] = `
"<svg width=\\"88.6\\" height=\\"20\\" viewBox=\\"0 0 886 200\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <linearGradient id=\\"a\\" x2=\\"0\\" y2=\\"100%\\">
    <stop offset=\\"0\\" stop-opacity=\\".1\\" stop-color=\\"#EEE\\"/>
    <stop offset=\\"1\\" stop-opacity=\\".1\\"/>
  </linearGradient>
  <mask id=\\"m\\"><rect width=\\"886\\" height=\\"200\\" rx=\\"30\\" fill=\\"#FFF\\"/></mask>
  <g mask=\\"url(#m)\\">
    <rect width=\\"368\\" height=\\"200\\" fill=\\"#555\\"/>
    <rect width=\\"518\\" height=\\"200\\" fill=\\"#3C1\\" x=\\"368\\"/>
    <rect width=\\"886\\" height=\\"200\\" fill=\\"url(#a)\\"/>
  </g>
  <g fill=\\"#fff\\" text-anchor=\\"start\\" font-family=\\"Verdana,DejaVu Sans,sans-serif\\" font-size=\\"110\\">
    <text x=\\"60\\" y=\\"148\\" textLength=\\"268\\" fill=\\"#000\\" opacity=\\"0.25\\">build</text>
    <text x=\\"50\\" y=\\"138\\" textLength=\\"268\\">build</text>
    <text x=\\"423\\" y=\\"148\\" textLength=\\"418\\" fill=\\"#000\\" opacity=\\"0.25\\">passing</text>
    <text x=\\"413\\" y=\\"138\\" textLength=\\"418\\">passing</text>
  </g>
  
</svg>"
`;

exports[`action-badges GET / returns the homepage 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"UTF-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
  <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
  <title>Action Badges</title>
  <link rel=\\"stylesheet\\" href=\\"https://unpkg.com/@primer/css@12.1.0/build/build.css\\" />
  <style>
    html, body, main { height: 100% }
  </style>
</head>
<body>
  <main class=\\"text-center width-full d-flex flex-items-center flex-column flex-justify-center\\">
    <div class=\\"Box border-gray-light p-3 box-shadow-extra-large text-center markdown-body\\">
      <h1 class=\\"mb-3\\">Action Badges</h1>
      <pre><code>![my badge](PATH/JasonEtco/jasonet.co)</code></pre>
      <pre><code>![my badge](PATH/JasonEtco/jasonet.co?workflow=test-my-code)</code></pre>
    </div>

    <p class=\\"note mt-3\\">Made <a href=\\"https://github.com/JasonEtco/action-badges\\">on GitHub</a> by <a href=\\"https://jasonet.co\\">@JasonEtco</a>.</p>
  </main>

  <script>
    const code = document.querySelectorAll('code')
    code.forEach(c => c.textContent = c.textContent.replace('PATH', window.location.origin))
  </script>
</body>
</html>
"
`;
